client/src/modules/list/components/Items/ListItem/ListItem.jsx

```diff
@@ -130,17 +147,45 @@ class ListItem extends PureComponent {
 
   handleItemDescription = value => this.setState({ itemDescription: value });
 
+  handleItemCloning = event => {
+    event.stopPropagation();
+    const {
+      cloneItem,
+      data: { _id: itemId },
+      match: {
+        params: { id: listId }
+      }
+    } = this.props;
+
+    this.setState({ pending: true });
+    const abortableCloning = makeAbortablePromise(cloneItem(listId, itemId));
+    this.appendPendingPromise(abortableCloning);
+
+    return abortableCloning.promise
+      .then(() => {
+        this.setState({ pending: false });
+        this.removePendingPromise(abortableCloning);
+      })
+      .catch(err => {
+        if (!(err instanceof AbortPromiseException)) {
+          this.setState({ pending: false });
+          this.removePendingPromise(abortableCloning);
+        }
+      });
```

Comment:
why this have to be done this way with 'abortable promises'
can it just be done like in any other actions ??

`cloneItem(listId, itemId).then(()=> //do something).catch(err)`

Reply:
If cloning the last item causes that the items count is above the limit of displayed items, this cloned item is being unmounted. Since there is still async action in progress we get warning in the browser. To prevent it I reject pending promises in componentWillUnmount. We want to do these actions only when the component is mounted so we have to check: if we created AbortPromiseException we know that component is unmounted and these actions shouldn't be invoked.